src/game.cpp

```diff
@@ -6118,17 +6118,28 @@ void game::peek( const tripoint &p )
     u.moves -= 200;
     tripoint prev = u.pos();
     u.setpos( p );
+    const bool is_same_pos = u.pos() == prev;
+    const bool is_standup_peek = is_same_pos && u.is_crouching();
```

Comment:
Could add action type check option here if base integration is too intrusive and seen as limiting player control.